test/steps/image_registry.go

```diff
@@ -70,6 +76,23 @@ func (data *Data) localServiceBuiltByMavenWithProfileAndDeployedToRuntimeRegistr
 	return nil
 }
 
+// Returns true if the image has to be built
+func needToBuildImage(namespace, imageTag string) bool {
+	switch config.GetImageCacheMode() {
+	case config.UseImageCacheAlways:
+		return false
```

Comment:
So, if we set image cache mode to always and the image is not built, it will fail. However, this is fine and expected, it would be a wrong setup rather.

Reply:
Right.